integration/testcmd/gotest/uinit/gotest.go

```diff
@@ -37,12 +39,50 @@ func walkTests(testRoot string, fn func(string, string)) error {
 	})
 }
 
+var (
+	coverprofile = flag.String("coverprofile", "", "Filename to write coverage data to")
+)
+
+func AppendFile(srcFile, targetFile string) error {
+	cov, err := os.Open(srcFile)
+	if err != nil {
+		return err
+	}
+	defer cov.Close()
+
+	out, err := os.OpenFile(targetFile, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0644)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
```

Comment:
Check the error returned by out.Close(). Since the file is being written to, if `Close` fails with an error, you might lose data.

Reply:
I thought Close() only returns an error if the file was already closed, so how does that cause data loss?

Reply:
Yes, see `man close`:

```
   Dealing with error returns from close()
       A careful programmer will check the return value of close(),  since  it
       is  quite possible that errors on a previous write(2) operation are re‐
       ported only on the final close() that releases the open  file  descrip‐
       tion.   Failing  to check the return value when closing a file may lead
       to silent loss of data.  This can especially be observed with  NFS  and
       with disk quota.
```

This only applies to file descriptors being written to. Read-only file descriptors cannot lose data.

Reply:
This still needs to be fixed.